Makefile

```diff
@@ -44,13 +44,20 @@ setup:
 	fi \
 	
 	
-	pip install cfn-lint pre-commit flake8 checkov==1.0.580
+	pip install cfn-lint pre-commit flake8 flake8-print checkov==1.0.580
 	pre-commit install
 
 run:
 	@echo Running pre-commit hook validation using $(HOOK_CONFIG_FILE)...
 	pre-commit clean
-	pre-commit run -c $(HOOK_CONFIG_FILE) --all-files
+
+	if [[ ! -z "$$EVENT_NAME" ]] && [[ "$$EVENT_NAME" == NIGHTLY ]]; then \
+		echo "Running in Nightly Mode"; \
+		pre-commit run -c $(HOOK_CONFIG_FILE) --all-files;\
+	else \
+		pre-commit run -c $(HOOK_CONFIG_FILE) --from-ref origin/master --to-ref HEAD;\
+	fi \
```

Comment:
@vishnu-narayanan 
1. What if I wanted to re-trigger a build from master with event name MANUAL and wanted to run on all files ?
2. What abour MERGE event ? this also should cover all files.

Reply:
Why is the CICD job different per event?

If I have a job that deploys a VPC for example. 

Why would I want that job to be different if it's nightly or manual? 

If you have jobs that are different based on the event then it needs to be a new job I think.

Reply:
@hanumantharaomvl 
> 1. What if I wanted to re-trigger a build from master with event name MANUAL and wanted to run on all files ?
Yeah, that is a valid question. The alternative is also true right, like if one wants to run MANUAL event with only changes from their branch. At the end of the day, we can decide what we want as a team and implement that way since this is opinionated. 

>2. What abour MERGE event ? this also should cover all files.
Yeah, you are right. I missed this event.

Reply:
@shadycuz 

> Why is the CICD job different per event?
If I have a job that deploys a VPC for example.
why would I want that job to be different if it's nightly or manual?
If you have jobs that are different based on the event then it needs to be a new job I think.


Nightly is supposed to run against our master branch to prevent regressions. Manual invocations are an opinionated topic. As I replied in the comment above, the implementation can be what we want as a team.

Reply:
Maybe for the linter as an expection. 

The problem is people will see this and duplicate it. 

A Jenkins job is a recipe to make X. Not [X,Y,Z]. 

Just keep that in mind for the future that the behavior of a job should be mostly consistent. We had this problem in the past where people tried to build a job that handled all the different use cases and it ended up unreadable and using a dozen large nested if/else statements....

See https://github.com/Flux7Labs/template-repo-jenkins-jobs/blob/991bbb65f7f141b7a620d132de4735b1deb9ab48/jobs/ami-factory/bake-ami/Jenkinsfile